### DISCUSSION TITLE
A better way to check if a user is followed by a couple hundred users

### DISCUSSION THREAD

**Discussion by Esoiya** [tweepy/tweepy#1593] [Questions] [answered]
I have a list of around 2000 twitter users and another lost of around 300 key users. 

For each of the 2000 twitter users, I want to be able to efficiently check if they are each followed by one of the 300 key users.

My original implementation:
```python
friendship_score = {}

for user in twitter_users:
    for target in key_users:
           follow = api.show_friendship(source_screen_name=user, target_id=target)[0].followed_by
           friendship_score[user].append(follow)
```

However, this is limited by [the rate 15/15min](https://developer.twitter.com/en/docs/twitter-api/v1/rate-limits). Consequently, it is far too time consuming.

Another approach I took, involved using the `Cursor`object:
``` python
for user in twitter_users:

   flist = []

    try:
        for page in tw.Cursor(api.followers_ids, screen_name=user, count=5000).page(5):
            flist.extend(page)
        
    except:
        continue
    
    for target in user_ids:
        
        follow = 0
        
        if target in flist:
            follow = 1

        friendship_score[user].append(follow)
```

However, it's also very time consumable if there are hundreds or even thousands of friends (followers, blockers, etc.).

`API.get_followers_ids()` returns a list of 5000 ids of user, once at most, per 15 minutes (5000/15min).

Is there a more efficient way to complete my task, that can be implemented in `tweepy`?

**Comment by Harmon758** [comment, marked as answer]
`API.get_followers_ids` does not exist. If you mean [`API.followers_ids`](https://tweepy.readthedocs.io/en/latest/api.html#tweepy.API.followers_ids), the [GET followers/ids](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-followers-ids) endpoint that that method uses allows you to retrieve up to a maximum of 5,000 IDs per request and has a rate limit of 15 requests / 15-min. window. This allows you to retrieve up to 75,000 IDs / 15 min., not 5,000.

Alternatively, you can use the [GET friends/ids](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/follow-search-get-users/api-reference/get-friends-ids) endpoint with [`API.friends_ids`](https://tweepy.readthedocs.io/en/latest/api.html#tweepy.API.friends_ids) to retrieve who the list of 300 users are following and cross-reference with the list of 2,000 users. That endpoint has the same rate limit and maximum number of IDs per request, so you should be able to retrieve up to 75,000 IDs / 15 min. with it as well. This would be faster if the number of users the list of 300 users are following is less than the number of users following the list of 2,000 users. It also decreases the minimum number of requests required (one for each user).

**Reply by Esoiya** [reply]
Yes that's what I meant & thank you, that method does seem like a better one.
